modules/table/src/test/java/org/apache/ignite/internal/table/distributed/replication/PartitionReplicaListenerTest.java

```diff
@@ -704,6 +705,30 @@ public void testTxStateReplicaRequestCommitState() throws Exception {
         assertTrue(readTimestamp.compareTo(txMeta.commitTimestamp()) > 0);
     }
 
+    @Test
+    public void testExecuteRequestOnFinishedTx() {
+        UUID txId = newTxId();
+
+        txStateStorage.put(txId, new TxMeta(ABORTED, singletonList(grpId), null));
+        txManager.updateTxMeta(txId, old -> new TxStateMeta(ABORTED, null, null, null));
+
+        BinaryRow testRow = binaryRow(0);
+
+        assertThat(doSingleRowRequest(txId, testRow, RequestType.RW_INSERT), willThrowFast(TransactionException.class));
+    }
```

Comment:
BTW I believe that it's fine to have finished even it's finishing within given context, what do you think?